File: CourseraAlgorithmicToolbox Resources/week3_greedy_algorithms/2_maximum_value_of_the_loot/by_learners/fractional_knapsack.rb
#!/usr/bin/env ruby
# by Andronik Ordian

def get_optimal_value(capacity, weights, values)
  value = 0.0
  # write your code here
  value
end

if __FILE__ == $0
  data = STDIN.read.split().map(&:to_i)
  n, capacity = data[0,2]
  values = data.values_at(*(2..2*n).step(2))
  weights = data.values_at(*(3..2*n+1).step(2))

  answer = get_optimal_value(capacity, weights, values)
  puts "#{'%.4f' % answer}"
end

File: Resources/Coursera/AlgorithmicToolbox/week3_greedy_algorithms/2_maximum_value_of_the_loot/FractionalKnapsack.kt
import java.util.*

fun getOptimalValue(capacity: Int, values: IntArray, weights: IntArray): Double {
    //write your code here
    return 0.0
}

fun main(args: Array<String>) {
    val scanner = Scanner(System.`in`)
    val n = scanner.nextInt()
    val capacity = scanner.nextInt()
    val values = IntArray(n)
    val weights = IntArray(n)
    for (i in 0 until n) {
        values[i] = scanner.nextInt()
        weights[i] = scanner.nextInt()
    }
    println(getOptimalValue(capacity, values, weights))
}